pkg/collector/corechecks/snmp/internal/checkconfig/config.go

```diff
@@ -217,27 +217,35 @@ type CheckConfig struct {
 }
 
 // SetProfile refreshes config based on profile
-func (c *CheckConfig) SetProfile(profile string) error {
-	if _, ok := c.Profiles[profile]; !ok {
-		return fmt.Errorf("unknown profile `%s`", profile)
+func (c *CheckConfig) SetProfile(profileName string) error {
+	profileConf := c.Profiles.GetProfile(profileName)
+	if profileConf == nil {
+		return fmt.Errorf("unknown profile `%s`", profileName)
 	}
-	log.Debugf("Refreshing with profile `%s`", profile)
-	tags := []string{"snmp_profile:" + profile}
-	definition := c.Profiles[profile].Definition
-	c.ProfileDef = &definition
-	c.Profile = profile
+	log.Debugf("Refreshing with profile `%s`", profileName)
+	c.AutoDetectProfileDef = nil
+	c.Profile = profileName
 
 	if log.ShouldLog(seelog.DebugLvl) {
-		profileDefJSON, _ := json.Marshal(definition)
-		log.Debugf("Profile content `%s`: %s", profile, string(profileDefJSON))
+		profileDefJSON, _ := json.Marshal(profileConf.Definition)
+		log.Debugf("Profile content `%s`: %s", profileName, string(profileDefJSON))
 	}
+	c.RebuildMetadataMetricsAndTags()
+	return nil
+}
 
-	if definition.Device.Vendor != "" {
-		tags = append(tags, "device_vendor:"+definition.Device.Vendor)
```

Comment:
I moved this tag computation to `RebuildMetadataMetricsAndTags`, so that there's one call that rebuilds all the cached values.